src/WrappedMToken.sol

```diff
@@ -504,23 +504,52 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
 
         if (amount_ == 0) return;
 
-        if (sender_ == recipient_) {
-            uint240 balance_ = _accounts[sender_].balance;
+        Account storage senderInfo_ = _accounts[sender_];
+        Account storage recipientInfo_ = _accounts[recipient_];
+
+        bool senderIsEarner_ = senderInfo_.isEarning;
+        bool recipientIsEarner_ = recipientInfo_.isEarning;
+
+        if (senderIsEarner_ != recipientIsEarner_) {
+            senderIsEarner_
+                ? _subtractEarningAmount(sender_, amount_, currentIndex_)
+                : _subtractNonEarningAmount(sender_, amount_);
 
-            if (balance_ < amount_) revert InsufficientBalance(sender_, balance_, amount_);
+            recipientIsEarner_
+                ? _addEarningAmount(recipient_, amount_, currentIndex_)
+                : _addNonEarningAmount(recipient_, amount_);
 
             return;
         }
 
-        // TODO: Don't touch globals if both are earning or not earning.
+        uint240 senderBalance_ = senderInfo_.balance;
+
+        if (senderBalance_ < amount_) revert InsufficientBalance(sender_, senderBalance_, amount_);
+
```

Comment:
```suggestion
 unchecked {
                senderInfo_.balance = senderBalance_ - amount_;
                recipientInfo_.balance += amount_;
            }
    if (!senderIsEarner_) return;
    
    unchecked {
                           senderInfo_.earningPrincipal = earningPrincipal_ - principal_;
                        recipientInfo_.earningPrincipal = UIntMath.safe112(uint256(recipientInfo_.earningPrincipal) + principal_);
            }
```

Reply:
consider the following adjustment if you like. 

Code will benefit from short comments mentioning each transfer type above section like 

```earner --> non-earner or non-earner --> earner```
```non-earner --> non-earner```
```earner --> earner```